### ISSUE TITLE
src/ files changed, but can not live-reload

### ISSUE THREAD

**Issue by allan2coder** [transitive-bullshit/create-react-library#44]
I have seen "anytime you make a change to your library in src/ or to the example app's example/src, create-react-app will live-reload" this words in readme. But It can not live-reload, when I run `npm start` in src/.

**Comment by transitive-bullshit** [comment]
Hmm, this functionality is working correctly for others. Would you mind providing a link to your source?

Generally, you run `npm start` from the project root, which starts rollup in watch mode.

Then, you'd run `npm start` from `example/` and CRA should automatically live-reload any changes you make to the example OR library `src/` on-the-fly.

**Comment by transitive-bullshit** [comment]
Also, what version of node / npm / and which platform are you using?

**Comment by allan2coder** [comment]
my npm version is 5.6.0, node version is v10.2.1.
![image](https://user-images.githubusercontent.com/14307551/43363334-3a0d2018-9335-11e8-9a9f-b03e4911a919.png)

**Comment by transitive-bullshit** [comment]
I see no reason why the live-reload shouldn't be working for you. Are you able to provide a link to the project's source?

**Comment by allan2coder** [comment]
For example, I have change the text in src/index.js, but in browser, it no changed.as following.
I will provide a link, but wait a moment.
Thank you.
![image](https://user-images.githubusercontent.com/14307551/43363797-5e122332-933f-11e8-8ed1-8fdb00dd5523.png)
![image](https://user-images.githubusercontent.com/14307551/43363819-c366a83e-933f-11e8-8ace-02655c1f087e.png)

**Comment by chrisvxd** [comment]
Another possibility that affected me:

If you've cleared out and reinstalled `node_modules` inside `example` at any point, npm will use a copied module rather than the symlinked one, which will break the live reload as CRA can't detect any local changes. To fix this, go into the `example` directory and run `yarn link your-package-name`.

@transitive-bullshit what do you think about adding a `postinstall` hook to the example project to automatically symlink after the install?

**Comment by allan2coder** [comment]
@chrisvxd  if u want to develop a ui library to NPM. You can try my project:https://github.com/allan2coder/react-npm-publish

**Comment by transitive-bullshit** [comment]
@chrisvxd that's a great idea, though more recent versions of CRL don't use symlinking for the example, but rather reference `file:..` as the dependency. This removes the need for remembering to symlink at all as the example will always reference the local version of the src module.

**Comment by chrisvxd** [comment]
@transitive-bullshit I've done some digging, and the issue I was encountering was by running `yarn` (v1.7.0) inside the `example` directory. Running `npm` (v5.6.0) in `example` fixes it. Turns out `yarn` _copies_ the local directory, whilst `npm@5` _symlinks_ it. `npm@2-4` also copied it. [See `npm@5` changelog](https://blog.npmjs.org/post/161081169345/v500).

Can reproduce locally on a clean project when configuring it to use yarn.

Perhaps symlinking manually would be a safer bet, since people might use older versions of npm, or yarn.

I'm not sure if this is the cause of issue that the OP is referring to (@allan2coder does this help?), so let me know if you want me to swing this into a new issue.

**Comment by transitive-bullshit** [comment]
@chrisvxd thanks again for that breakdown; I had made the `link:..` to `file:..` change hastily and didn't have proper test coverage for live reload.

I just released `v2.5.0` which should fix this issue. I ended up enforcing `npm>=5` and using `link:..` for yarn and `file:..` for npm. I still need to add tests for this issue that will test multiple versions of npm and yarn, but I wanted to get this fix out quickly since I wasn't aware that it had broken.

Thanks everyone!
